hazelcast/src/main/java/com/hazelcast/config/properties/PropertyTypeConverter.java

```diff
@@ -29,7 +29,10 @@
     STRING {
         @Override
         public Comparable convert(Comparable value) {
-            return value.toString();
+            if (value != null) {
```

Comment:
It is a flavor thing, but the initial code of my method is the unhappy stuff so that is out of the way when the real work begins. I normally do something like:

if(value == null){.....}

And in case of Hazelcast, we use the Preconditions. This is cleaner and less lines. I would also include the reason why you can't convert to string btw. Now a user might be a bit baffled.